analysis/analysis.js

```diff
@@ -0,0 +1,163 @@
+            var i =0;
+            for(i=0; i<100; i++){
+              frame[i]=rideFile.data[i];
+            }
+            while(i<(rideFile.data.length)){
+              if((frame[frameCenter].geo.speed*3.6) <=10 && Math.abs(frame[frameCenter].acc.z > 1.8 )){
+                console.log(++count);
+                var result = analyzeFrame(rideId, frame);
+                if(result){
+                  violations.push(result);
+                }
+              }else if((frame[frameCenter].geo.speed*3.6) <=15 && Math.abs(frame[frameCenter].acc.z > 2.4 )){
+                console.log(++count);
+                var result = analyzeFrame(rideId, frame);
+                if(result){
+                  violations.push(result);
+                }
+              }else if((frame[frameCenter].geo.speed*3.6) <=20 && Math.abs(frame[frameCenter].acc.z > 3.1 )){
+                console.log(++count);
+                var result = analyzeFrame(rideId, frame);
+                if(result){
+                  violations.push(result);
+                }
+              }else if((frame[frameCenter].geo.speed*3.6) <=25 && Math.abs(frame[frameCenter].acc.z > 4 )){
+                console.log(++count);
+                var result = analyzeFrame(rideId, frame);
+                if(result){
+                  violations.push(result);
+                }
+              }else if((frame[frameCenter].geo.speed*3.6) > 25 && Math.abs(frame[frameCenter].acc.z > 4.7 )){
+                console.log(++count);
+                var result = analyzeFrame(rideId, frame);
+                if(result){
+                  violations.push(result);
+                }
+              };
+              frame.push(rideFile.data[i])
+              frame.shift();
+              i++;
+            }
+          }else {
+            console.log("Ride is too short!");
+            return;
+          }
+          var analysisEndTime = Date.now();
+          var analysisInfo = JSON.stringify({
+            'status': 'completed',
+            'version': 1,
+            'startedAt': analysisStartTime,
+            'endedAt': analysisEndTime,
+            'violations': violations
+          });
+          var putQuery = {
+            'rideId': rideId,
+            'analysisInfo': analysisInfo
+          };
+            request.put({
+              url:'http://localhost:3000/rideInfo/' + rideId,
+              body: putQuery,
+              json: true
+            },
+            function(err,httpResponse,body){
+              if(httpResponse.body.success == true){
+                console.log("Succesfully updated for rideId" + httpResponse.body.rideId);
+              } else {
+                console.log("Failed to update ");
+              }
+            if(err){
+              console.log(err);
+            }
+          });
+        };
+      }
+    });
+  }
+  analysisInprogress = false;
+};
+
+module.exports.analyzeRide = analyzeRide;
+
+analyzeRide();
```

Comment:
Dont call function inside the module. This will make the module execute the code just by including it.